parlai/tasks/cmu_dog/agents.py

```diff
@@ -0,0 +1,370 @@
+    convo_files = _all_split_datafiles(opt)
+
+    for fname in convo_files:
+        with PathManager.open(os.path.join(datapath, f"conversations/{fname}")) as f:
+            data = json.load(f)
+        convos += _collect_convo_text(data)
+
+    return RareWordF1Calculator(' '.join(convos), top_p=0.5)
+
+
+class CMUDocumentGroundedConversationsTeacher(DialogTeacher):
+    """
+    CMU Document Grounded Conversations Dataset (aka CMU_DoG)
+
+    Paper: https://arxiv.org/pdf/1809.07358.pdf
+    Source: https://github.com/festvox/datasets-CMU_DoG
+    """
+
+    def __init__(self, opt: Opt, shared: TShared = None):
+        opt = copy.deepcopy(opt)
+        self.delimiter = opt.get('delimiter', '\n')
+        split_type = SplitType(opt.get("cmu_dog_split_type"))
+        if split_type == SplitType.ORIGINAL:
+            logger.warning(
+                "`original` split type contains duplicate conversations across train, "
+                "valid, and test. See https://github.com/festvox/datasets-CMU_DoG/issues/2 "
+                "for more detail."
+            )
+        opt['datafile'] = _datafile(
+            split=DatatypeHelper.fold(opt['datatype']), split_type=split_type
+        )
+        super().__init__(opt, shared)
+        if shared:
+            self.rare_word_f1 = shared['rare_word_f1']
+        else:
+            self.rare_word_f1 = _build_rare_word_f1(opt)
+
+    @classmethod
+    def add_cmdline_args(
+        cls, parser, partial_opt: Optional[Opt] = None
+    ) -> ParlaiParser:
+        super().add_cmdline_args(parser, partial_opt=partial_opt)
+        cmu_dog = parser.add_argument_group("CMU Document Grounded Conversations")
+        cmu_dog.add_argument(
+            "--cmu-dog-rating",
+            type=int,
+            default=[1, 2, 3],
+            choices=[1, 2, 3],
+            nargs="+",
+            help='The higher the number, the better quality the conversation. '
+            'For each rating, the number of conversations is as follows: 1-1443, 2-2142, 3-527',
+        )
+        cmu_dog.add_argument(
+            "--cmu-dog-only-with-knowledge",
+            type=bool,
+            default=True,
+            help="Optionally train only the sides of the conversation that have access to knowledge.",
+        )
+        cmu_dog.add_argument(
+            "--cmu-dog-multi-msg-delimiter",
+            type=str,
+            default=" ",
+            help="When one agent is to send multiple messages in a row, they will be concatenated with this delimiter.",
+        )
+        cmu_dog.add_argument(
+            "--cmu-dog-fact-delimiter",
+            type=str,
+            default=";",
+            help="When a section of the knowledge contains multiple facts, they will be concatenated with this delimiter.",
+        )
+        cmu_dog.add_argument(
+            "--cmu-dog-include-knowledge-keys",
+            type=str,
+            default='cast,critical_response,director,genre,introduction,movieName,rating,year',
```

Comment:
should we make `movieName` be the default here? or at least the `recommended`? as at least for our future experimentation we'll like to use that split.

alternatively, maybe we can provide a teacher, like `MovieNameTeacher`, which sets this arg for you (so you can do something like `parlai dd -t cmu_dog:movie_name` and it'll work appropriately)